### PR TITLE
Adding some scripts to facilitate CI builds (and other helpers)

### PR DESCRIPTION
**Pull request by CalvinAllen**

### CODE REVIEW

File: build/src/vs/Build.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+	try {
+		execSync("npm run build:ci", { stdio: "inherit", cwd: `${vsRootPath}` });
+	} catch (error) {
+		console.error("Error executing command:", error);
+		process.exit(1);
+	}
+
+	try {
+		execFileSync(
```
Comment (bcanzanella) [inline build/src/vs/Build.ts:17]: 🔥

File: build/src/vs/PreBuild.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	}
+
+	if (!fs.existsSync(`${vsRootPath}/src/CodeStream.VisualStudio.Vsix.x64/dist/agent`)) {
+		fs.mkdirSync(`${vsRootPath}/src/CodeStream.VisualStudio.Vsix.x64/dist/agent`, {
+			recursive: true
+		});
+	}
+
+	fs.copyFileSync(
+		`${supplementalSoftwarePath}/node/node-v18.15.0-win-x64/node.exe`,
```
Comment (bcanzanella) [inline build/src/vs/PreBuild.ts:30]: hoist that up or make a config option for that path?

**Review by bcanzanella — APPROVED** [review]
LGTM

**Comment by CalvinAllen** [comment]
VS was a success - 

https://teamcity.cdstrm.dev/buildConfiguration/VSV2_Ci/27541?expandBuildDeploymentsSection=false&hideTestsFromDependencies=false&hideProblemsFromDependencies=false&pluginCoverage=true

Also, somehow, I fixed the testing issue 🤷🏻

**Comment by bcanzanella** [comment]
ship it.

File: build/Build.ts
Comment (dsellarsnr) [inline build/Build.ts:1]: there are a few .ts files not under `src/` - can they be moved there? btw i'd like to update `track.js` to ts and move it there as well if that is ok.

Comment (CalvinAllen) [reply build/Build.ts:1]: Shouldn't be an issue, the structure/folders/files morphed quite a bit trying to figure out the "best" approach

**Pull request merged**